File: CPP/modelo/TorreArquera.cpp
#ifndef TORRE_ARQUERA_CPP
#define TORRE_ARQUERA_CPP

#include "Torre.cpp"

class TorreArquera : public Torre {
private:
    static const int DANO_BASE = 8;

public:
    TorreArquera(std::string nombre, int posicion) : Torre(nombre, 4, posicion) {}

    int calcularDano() const override { return DANO_BASE; }
    std::string tipo() const override { return "TorreArquera"; }
};

#endif

File: Java/src/modelo/TorreArquera.java
package modelo;

public class TorreArquera extends Torre {
    private static final int DANO_BASE = 8;

    public TorreArquera(String nombre, int posicion) {
        super(nombre, 4, posicion);
    }

    @Override
    public int calcularDano() {
        return DANO_BASE;
    }
}
